melodies_monet/driver.py

```diff
@@ -187,12 +201,72 @@ def open_obs(self, time_interval=None, control_dict=None):
             return
         
         self.add_coordinates_ground() # If ground site then add coordinates based on yaml if necessary
+            
         self.mask_and_scale()  # mask and scale values from the control values
         self.rename_vars() # rename any variables as necessary 
         self.sum_variables() 
         self.resample_data()
         self.filter_obs()
 
+        if self.extra_calc is not None:
+            print("Performing extra calculations for obs...")
+
+            # potential temperature is the only calculated function supported for the obs. 
+            # variables such as dewpoint / relh are a) directly available or b) uncalculable without
+            # further building out the metcalc util file. Obs do not normally have specific humidity. 
+            # If these commented out functs are wanted in the future, other metpy libraries to calc 
+            # relh / dewpoint need to be used. 
+
+            # if "dewpoint" in self.extra_calc:
+            #     print("Calculating observed Dewpoint...")
+            #     from .util.metcalc import dewpoint # import functions from the util.metcalc file
+                
+            #     varmap = self.extra_calc["dewpoint"]
+            #     self.obj=dewpoint(self.obj, varmap = varmap)
+            #     #print(self.obj)
+
+            # if "rel_hum" in self.extra_calc:
+            #     print("Calculating observed relative humidity...")
+            #     from .util.metcalc import relh
+
+            #     varmap = self.extra_calc["rel_hum"]
+            #     self.obj=relh(self.obj, varmap = varmap)
+            #     #print(self.obj)
+
+            # if "windspeed" in self.extra_calc:
+            #     print("Calculating observed windpseed...")
+            #     from .util.metcalc import wspd
+
+            #     varmap = self.extra_calc["windspeed"]
+            #     self.obj=wspd(self.obj, varmap = varmap)
+            #     #print(self.obj)
+                
+            # if "winddir" in self.extra_calc:
+            #     print("Calculating observed wind direction...")
+            #     from .util.metcalc import wdir
+
+            if "ptemp_obs" in self.extra_calc:
+                print("Calculating observed potential temperature...")
+                from .util.metcalc import ptemp
+            
+                varmap = self.extra_calc["ptemp_obs"]
+                self.obj = ptemp(
+                    self.obj,
+                    varmap=varmap,
+                    output_key="ptemp_obs",
+                    default_keys={"pressure": "pressure_obs", "temperature": "temperature_C"}
+                )
+    
+            # uncomment once working
```

Comment:
I'm fine keeping this in but commented as a placeholder for how we would add this in in the future.